uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php

```diff
@@ -250,13 +251,16 @@ private static function saveLeaveBalanceChange(
   ) {
     $balanceChangeTypes = array_flip(LeaveBalanceChange::buildOptions('type_id'));
 
-    $proRata = $calculation->getProRata();
+    //The original pro-rata calculation already factors in public holidays
+    //since public holiday balance changes are saved differently, we need to deduct it from the pro rata
+    $adjustedProRata = $calculation->getProRata() - $calculation->getNumberOfPublicHolidaysInEntitlement();
+
 
     LeaveBalanceChange::create([
       'type_id' => $balanceChangeTypes['Leave'],
       'source_id' => $periodEntitlement->id,
       'source_type' => LeaveBalanceChange::SOURCE_ENTITLEMENT,
-      'amount' => $proRata
+      'amount' => $adjustedProRata
```

Comment:
Since this is the only place were we're using this variable, we can ditch it and just do
```php
'amount' => $calculation->getProRata() - $calculation->getNumberOfPublicHolidaysInEntitlement()
```

But please keep the comment